aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -99,7 +99,8 @@ Tensor _dim_arange(const Tensor& like, int64_t dim) {
 
 // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ empty ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Tensor empty_cpu(IntArrayRef size, const TensorOptions& options_, c10::optional<c10::MemoryFormat> optional_memory_format) {
-
+  TORCH_CHECK((options_.dtype() != caffe2::TypeMeta::Make<std::complex<float>>()) &&
```

Comment:
probably easier to check that the dtype/scalartype isn't complex.

Reply:
hmm .is_complex() is only callable on a tensor and .is_complex a dtype attribute but only on the Python side so not sure how to do this except by creating a tensor and calling is_complex?

Reply:
https://github.com/pytorch/pytorch/blob/e3daf7018499fbcb52de4018543fd9a0306ccca6/c10/core/ScalarType.h#L307-L311

Reply:
updated